src/engine/JoinHelpers.h

```diff
@@ -76,21 +85,29 @@ inline std::variant<LazyInputView, MaterializedInputView> resultToView(
   return convertGenerator(result.idTables(), permutation);
 }
 
-using GeneratorWithDetails =
-    cppcoro::generator<ad_utility::IdTableAndFirstCol<IdTable>,
-                       CompressedRelationReader::LazyScanMetadata>;
+// Type alias for the general InputRangeTypeErasedWithDetails with specific
+// types
+using GeneratorWithDetails = ad_utility::InputRangeTypeErasedWithDetails<
+    ad_utility::IdTableAndFirstCol<IdTable>,
+    CompressedRelationReader::LazyScanMetadata>;
 
 // Convert a `generator<IdTable` to a `generator<IdTableAndFirstCol>` for more
 // efficient access in the join columns below.
 inline GeneratorWithDetails convertGenerator(
     Permutation::IdTableGenerator gen) {
-  co_await cppcoro::getDetails = gen.details();
-  gen.setDetailsPointer(&co_await cppcoro::getDetails);
-  for (auto& table : gen) {
+  // Capture the details before moving the generator
+  auto details = gen.details();
```

Comment:
This also doesn't quite work yet. The details are empty at this point, as the generator will only populate them once we consume it (which will happen lazily by the range you are using here.
What you have to  do is to pass the `GeneratorWithDetails` a reference or pointer to the details, with the semantics "you don't own your details, but you share it with your underlying range". 
Tell me if you need help with the implementation of this feature.